static/js/util/profile_edit.js

```diff
@@ -446,7 +446,7 @@ export function saveAndContinue(nestedValidationCallback, finalStep) {
   }
 
   let clone = Object.assign({}, profile, {
-    filled_out: !!finalStep
+    filled_out: profile.filled_out || finalStep
```

Comment:
If `finalStep` is ever not a boolean (for example if we omit the argument) then this value may end up being `undefined`, which will in turn cause the `PATCH` not to send it in the JSON at all, since `JSON.stringify` ignores undefined values.